File: curso-python/ex045.py
from time import sleep
from random import randint

opcoes = {0: "Pedra",1: "Papel",2: "Tesoura"}
regras = {
    "Pedra":"Tesoura",
    "Papel":"Pedra",
    "Tesoura":"Papel"
}
print("Suas opções:")
for chave, valor in opcoes.items():
    print(f"[ {chave} ] {valor}")

jogador = int(input("qual a sua jogada: "))
computador = randint(0,2)

print("JO")
sleep(0.8)
print("KEN")
sleep(0.8)
print("PO!")

# Exibe as jogadas
print(f"Jogador jogou: {opcoes[jogador]}")
print(f"Computador jogou: {opcoes[computador]}")

if opcoes[jogador] == opcoes[computador]:
    print("deu empate")

# se a opção do jogador por exemplo for tesoura, a regra diz que tesoura ganha do papel, se o computador for igual ao papel, então jogador vence
elif regras[opcoes[jogador]] == opcoes[computador]:
    print("jogador venceu")
else:
    print("computador venceu")

File: javascript-exercicios/ex045.js
const prompt = require("prompt-sync")()

let opcoes = {
    0: "Pedra",
    1: "Papel",
    2: "Tesoura"
} 
let regras ={
    "Pedra":"Tesoura",
    "Papel":"Pedra",
    "Tesoura":"Papel"
}

console.log("suas opções: ")
for (const chave in opcoes){
    console.log(`[ ${chave} ] ${opcoes[chave]}`)
}

const jogador = Number(prompt("qual a sua jogada: "))
const computador = Math.floor(Math.random() * 3);

console.log(`Jogador jogou: ${opcoes[jogador]}`);
  console.log(`Computador jogou: ${opcoes[computador]}`);

if (opcoes[jogador] === opcoes[computador]) {
    console.log("Deu empate");
} else if (regras[opcoes[jogador]] === opcoes[computador]) {
    console.log("Jogador venceu");
} else {
    console.log("Computador venceu");
}